### ISSUE TITLE
It seems that Mephisto always connects only to the sandbox version of Mechanical Turk

### ISSUE THREAD

**Issue by claudiogreco** [facebookresearch/Mephisto#714]
Hello,

It seems to me that my app built using Mephisto always connects only to the sandbox version of MTurk. Indeed, when I run:
```
python run.py mephisto/architect=heroku mephisto/provider=mturk_sandbox mephisto.provider.requester_name=REQUESTER_NAME conf=mturk
```
it works, generating links such as https://workersandbox.mturk.com/mturk/preview?groupId=GROUPID.
However, when I run:
```
python run.py mephisto/architect=heroku mephisto/provider=mturk mephisto.provider.requester_name=REQUESTER_NAME conf=mturk
```
(Basically, replacing "mturk_sandbox" with "mturk" in the "provider" parameter)
It seems to still use sandbox. Indeed, it generates links such as https://workersandbox.mturk.com/mturk/preview?groupId=GROUPID. Moreover, I cannot find the created task on the HIT Groups section of mturk.com when I login as a worker.

Could you please help me?

Thank you in advance.

**Comment by JackUrb** [comment]
Hi @claudiogreco - this is happening because Mephisto ultimately ends up selecting your chosen `REQUESTER_NAME` over the `provider` argument. The MTurk Sandbox and MTurk versions of a requester are two different things to Mephisto, and we key by name. 

This is why our quickstart guide has [a section](https://mephisto.ai/docs/guides/quickstart/#optional-set-up-mturk) around setting up both your `REQUESTER` and `REQUESTER_sandbox` requesters.

On our side, we can definitely do a better job in our config parsing to note when there's a mismatch between the requester _type_ provided and the type of a specified requester (if provided).

**Comment by claudiogreco** [comment]
Hi @JackUrb,

It seems to work thanks to your suggestions. Thank you so much!

**Comment by JackUrb** [comment]
I've added a warning to this end into our config parsing script [here](https://github.com/facebookresearch/Mephisto/pull/691/commits/cdaebdcbd1147cd3bb0f6707167c187969a1d508). It'll be included in our 1.0 release.

Thanks for bringing this to my attention!
